modules/apps/organizations/organizations-item-selector-web/src/main/java/com/liferay/organizations/item/selector/web/internal/display/context/OrganizationItemSelectorViewDisplayContext.java

```diff
@@ -109,10 +103,27 @@ public SearchContainer<Organization> getSearchContainer()
 		return _searchContainer;
 	}
 
+	private List<Organization> _filterOrganizations(
+			List<Organization> organizationList)
+		throws PortalException {
+
+		List<Organization> filteredList = new ArrayList<>();
+
+		for (Organization organization : organizationList) {
+			if (OrganizationPermissionUtil.contains(
+					PermissionThreadLocal.getPermissionChecker(),
+					organization.getOrganizationId(), ActionKeys.VIEW)) {
+
+				filteredList.add(organization);
+			}
+		}
+
+		return filteredList;
+	}
+
```

Comment:
I don't think this is the right solution because we are filtering after the search. The risk in this case is that the pagination breaks.
I'm asking page 1 from 1 to 10 and I receive 3 items and then there is a page 2 😕